### PR TITLE
docs(cdn-caching): Generate cache keys and toggle CDN providers

### PR DESCRIPTION
**Pull request by Harshith46**
This PR handles the documentation for
1. generating cache keys 
2. How cache headers are set
3. how to toggle between CDN providers

### CODE REVIEW

**Review by ags1773 — APPROVED** [review]

File: tutorial/cdn-caching.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+        header | value
+        --- | ---
+        "Edge-Control" | "public,maxage=60,stale-while-revalidate=150,stale-if-error=3600"
+        "Vary" | "Accept-Encoding"
+
+How Quintype handles toggle between different CDN providers ?
+
+- For now, we have support for `Cloudflare`, `Fastly`, and `Akamai`.
+- The default CDN provider is set as `Cloudflare`.
```
Comment (rashmirmittal) [inline tutorial/cdn-caching.md:129]: Mention Fastly is used as backup CDN if Cloudflare goes down

File: tutorial/cdn-caching.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+---
+
+# {{page.title}}
+
+_This tutorial was contributed by [Harshith](ttps://www.linkedin.com/in/harshith-raj-092ba4176)_
+
+How Quintype handles CDN caching?
+
+- Whenever a collection/story/author is created or updated, the bold handles cache/purging through the CDN.
+- When a collection/story/author is created or updated, the frontend gets updated with the latest data with the help of cache-keys.
```
Comment (rashmirmittal) [inline tutorial/cdn-caching.md:14]: This is still vague, mention "the frontend gets updated with the latest data with the help of cache-keys returned in the API for that collection/story/author"

File: tutorial/cdn-caching.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+   We concat the collection cache keys, story cache keys, section collection cache keys to the above-generated key. The front end gets updated after it sees the cache keys from the bold when the home collection gets updated and published.
+
+   Eg:
+
+   `< cache-tag: c/1166/87520, c/1166/96964, s/1166/5b10a46a, s/1166/185a3cb0, page/1166/home-page`
+
+   Note:
+
+   - We form the cache keys for nested collection by going from the top-level depth of the collections .i.e, for example, the home collection contains 2 collections and each of these collections has a story then based on the above example `c/1166/87520` is generated based on the first collection in home, `c/1166/96964`is generated based on the second collection in home, `s/1166/5b10a46a` is generated based on the story in the first collection and so on
```
Comment (rashmirmittal) [inline tutorial/cdn-caching.md:37]: Specify the default nesting depth, don't remember, its 1 or 2.

File: tutorial/cdn-caching.md (hunk 1 above)
Comment (Harshith46) [reply tutorial/cdn-caching.md:129]: sure

File: tutorial/cdn-caching.md (hunk 2 above)
Comment (Harshith46) [reply tutorial/cdn-caching.md:14]: sure

File: tutorial/cdn-caching.md (hunk 3 above)
Comment (Harshith46) [reply tutorial/cdn-caching.md:37]: sure

Comment (Harshith46) [reply tutorial/cdn-caching.md:37]: @rashmirmittal, is this required we are just mentioning how keys are being formed w.r.t nested collection

Comment (rashmirmittal) [reply tutorial/cdn-caching.md:37]: Required because there is a default depth and custom FEs can override the depth

**Review by rashmirmittal — APPROVED** [review]

**Pull request merged**